### PR TITLE
Fixes in GET /v2/types and GET /v2/types/T operations

### LINKED ISSUE #1833: APIv2: use of map for listing types ?
The current APIv2 uses a map to return the list of entitiy types using type names as key whereas others endpoints (subscriptions, registrations and entities) use an array of objects.
Why use a map only for types, it could have been used for registrations and subscriptions which also have unique IDs ?
Or why not always use array for consistency across the API ? Implementations of the API in strongly types languages can benefit from such consistency to support pagination (with elements count) in a more generic way.

### PR DESCRIPTION
**Pull request by fgalan**
Fixes #1833 and bullet 2 in #1636 

LGTM needed from:

@crbrox 
@kzangeli 

Be aware of and nice-to-have feedback:

@jmcanterafonseca 
@iariasleon

### CODE REVIEW

File: test/functionalTest/cases/1705_csub_cache_objects/csub_cache_string_filter_update.test [hunk 1]
```diff
@@ -287,7 +287,7 @@ Date: REGEX(.*)
 ====================================================
 POST http://localhost:9997/notify
 Content-Length: 461
-User-Agent: orion/1.0.0-next libcurl/7.19.7
+User-Agent: orion/REGEX(\d+\.\d+\.\d+.*)
```
Comment (fgalan) [inline test/functionalTest/cases/1705_csub_cache_objects/csub_cache_string_filter_update.test]: This was probably a lefover of a previous PR, which has passed unoticed until now.

NTC

File: doc/apiary/v2/fiware-ngsiv2-reference.apib [hunk 2]
```diff
@@ -1251,46 +1259,50 @@ Response code:
...
 
 ## Entity type [/v2/types/{entityType}]
 
 ### Retrieve entity type [GET /v2/types/{entityType}]
 
-The operation returns a JSON object with information about the type: `attrs` (the union set of attribute names along all
-the entities of such type) and `count` (the number of entities belonging to that type).
+The operation returns a JSON object with information about the type:
+
+* `attrs` : the union set of attribute names along all the entities of such type, represented in
```
Comment (kzangeli) [inline doc/apiary/v2/fiware-ngsiv2-reference.apib]: union set ... sounds a little strange ...   just 'set' or 'collection' ... ?
along -> along with
which -> whose

File: test/functionalTest/cases/1705_csub_cache_objects/csub_cache_string_filter_update.test (hunk 1 above)
Comment (kzangeli) [inline test/functionalTest/cases/1705_csub_cache_objects/csub_cache_string_filter_update.test]: So it seems.
Pretty sure it was a cut 'n paste from somewhere.
We probably have more of those ...

**Comment by iariasleon** [comment]
LGTM

File: doc/apiary/v2/fiware-ngsiv2-reference.apib [hunk 3]
```diff
@@ -1251,46 +1259,50 @@ Response code:
...
               "speed": {
-                "type": "none"
-              },
-              "fuel": {
-                "type": "gasoline"
+                "types": [ "none" ]
               },
               "fuel": {
-                "type": "diesel"
+                "types": [ "gasoline", "diesel" ]
```
Comment (crbrox) [inline doc/apiary/v2/fiware-ngsiv2-reference.apib]: Should it be sorted ? 
(diesel < gasoline)

File: doc/apiary/v2/fiware-ngsiv2-reference.apib (hunk 2 above)
Comment (fgalan) [inline doc/apiary/v2/fiware-ngsiv2-reference.apib]: Fixed in a35a7b9d44dfca9df550752e9b71937f1ab6850c

File: doc/apiary/v2/fiware-ngsiv2-reference.apib (hunk 3 above)
Comment (fgalan) [inline doc/apiary/v2/fiware-ngsiv2-reference.apib]: Not really... types in this array appear in the same orther the are retrieved from DB. Ordering them would have a computational cost and don't have too much value...

**Comment by crbrox** [comment]
LGTM

**Comment by kzangeli** [comment]
LGTM

**Pull request merged**